opm/input/eclipse/Schedule/Well/GridIndependentWellKeywordHandlers.cpp

```diff
@@ -0,0 +1,226 @@
+        std::vector<std::pair<double, double>> cell_md_and_tvd;
+        trajectory_segments.reserve(intersections.size());
+        cell_md_and_tvd.reserve(intersections.size());
+        const auto& ecl_grid = handlerContext.grid.get_grid();
+        for (const auto& intersection : intersections) {
+            const auto ijk = ecl_grid->getIJK(intersection.globCellIndex);
+            trajectory_segments.push_back({intersection.startMD, intersection.endMD, ijk});
+            double cell_md = 0.5 * (intersection.startMD + intersection.endMD);
+            double cell_tvd = wellPathGeometry->interpolatedPointAlongWellPath(cell_md)[2];
+            cell_md_and_tvd.emplace_back(cell_md, cell_tvd);
+        }
+        return std::pair {trajectory_segments, cell_md_and_tvd};
+    }
+
+
+    void
+    process_segments(HandlerContext& handlerContext,
+                     Opm::Well& well,
+                     const std::vector<external::WellPathCellIntersectionInfo>& intersections,
+                     const external::cvf::ref<external::RigWellPath>& wellPathGeometry,
+                     double diameter)
+    {
+        if (well.isMultiSegment()) {
+            // For now, no segments may be defined via WELSEGS, except for the top:
+            if (well.getSegments().size() > 1) {
+                const auto msg
+                    = fmt::format("   {} already defines segments with the WELSEGS keyword", well.name());
+                throw Opm::OpmInputError(msg, handlerContext.keyword.location());
+            }
+
+            auto [trajectory_segments, cell_md_and_tvd]
+                = get_segment_geometries(handlerContext, intersections, wellPathGeometry);
+            well.addWellSegmentsFromLengthsAndDepths(
+                cell_md_and_tvd, diameter, handlerContext.keyword.location());
+            auto new_connections = Compsegs::getConnectionsAndSegmentsFromTrajectory(
+                trajectory_segments, well.getSegments(), well.getConnections(), handlerContext.grid);
+            well.updateConnections(
+                std::make_shared<WellConnections>(std::move(new_connections)), false);
+            handlerContext.record_well_structure_change();
+        }
+    }
+
+
+    void
+    handleCOMPTRAJ(HandlerContext& handlerContext)
+    {
+        // Keyword WELTRAJ must be read first
+        std::unordered_set<std::string> wells_to_update;
+        external::cvf::ref<external::cvf::BoundingBoxTree> cellSearchTree {};
+
+        for (const auto& record : handlerContext.keyword) {
+            const auto wellNamePattern = record.getItem("WELL").getTrimmedString(0);
+            const auto wellnames = handlerContext.wellNames(wellNamePattern, false);
+
+            for (const auto& name : wellnames) {
+                auto well = handlerContext.state().wells.get(name);
+                auto connections = std::make_shared<WellConnections>(well.getConnections());
+
+                if (!well.getConnections().empty()) {
```

Comment:
this should be checked before making a copy through 
```c++
                auto connections = std::make_shared<WellConnections>(well.getConnections());
```

And since it is already in namespace Opm, you can remove all the usage of Opm:: in this file.